db/ens.go

```diff
@@ -495,12 +542,11 @@ func GetAddressForEnsName(name string) (address *common.Address, err error) {
 
 func GetEnsNameForAddress(address common.Address) (name *string, err error) {
 	err = ReaderDb.Get(&name, `
-	SELECT n.name 
-	FROM ens_addresses a
-	INNER JOIN ens_names n
-		on a.name_hash = n.name_hash
+	SELECT ens_name 
+	FROM ens
 	WHERE
-		a.address = $1;
-	`, address.Bytes())
+		address = $1 AND
+		is_primary_name
```

Comment:
You have added checks to `valid_to`, this looks better.

Your "Exactly it would switch to the second automatically - the user would need to set it again." confuses me though, it feels like it is missing a "not".
"Exactly it would switch to the second automatically" ... if that's the case, why would the user have to set it (what is it?) again? Is "it" the "primary.eth" i.e. do you mean that the user would have to buy said domain again?

The thing is:
If the user registers both names and the primary one expires, our system will not find the secondary. Is that correct?

Reply:
Sorry type ... is should say * it wouldn't switch to the second automatically*